### PR TITLE
feat(crux-ui): editor highlighting

### PR DESCRIPTION
**Pull request by m8vago**
feat: add editor input highlighting to version image editing and deployment editing.
feat: add editors list to relevant pages
feat: add focus based saving
fix: a couple of websocket related errors
refactor: add useRepatch() and replace useReducer() in some components
refactor: state handling in websocket edited pages

### CODE REVIEW

**Review by robot9706 — CHANGES_REQUESTED** [review]
Minor typo otherwise LGTM 😎

**Review by balanceee — APPROVED** [review]
+1

File: web/crux-ui/src/components/editor/multi-input.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+interface MultiInputProps extends Omit<DyoInputProps, 'id' | 'onFocus' | 'onBlur' | 'onChange'> {
+  id: string
+  disabled?: boolean
+  editorOptions: MultiInputEditorOptions
+  onPatch: (value: string) => void
+}
+
+const MultiInput = (props: MultiInputProps) => {
+  const { id, disabled, editorOptions, style: propsStyle, value, onPatch, ...forwaredProps } = props
```
Comment (peterambrus3) [inline web/crux-ui/src/components/editor/multi-input.tsx:13]: typo forwardedProps ?

**Review by peterambrus3 — CHANGES_REQUESTED** [review]
comment, otherwise lgtm

File: web/crux-ui/src/components/editor/multi-input.tsx (hunk 1 above)
Comment (m8vago) [reply web/crux-ui/src/components/editor/multi-input.tsx:13]: yap, ty!

**Review by robot9706 — APPROVED** [review]
LGTM 😎

**Review by peterambrus3 — APPROVED** [review]

**Pull request merged**